fmf/base.py

```diff
@@ -544,3 +547,58 @@ def node(reference):
                 "No tree node found for '{0}' reference".format(reference))
         # FIXME Should be able to remove .cache if required
         return found_node
+
+    def get_raw_data_node(self):
+        raw_data_node = self
+        id_list = list()
+        while True:
+            raw_data = raw_data_node.raw_data
+            if raw_data:
+                return raw_data_node, id_list
+            elif not raw_data_node.parent:
+                raise utils.RootError("Cannot find root node with raw_data, very strange")
+            else:
+                id_list.insert(0, "/" + raw_data_node.name.rsplit("/")[-1])
+                raw_data_node = raw_data_node.parent
+
+    def node_get_raw_data(self):
+        raw_data_node, id_list = self.get_raw_data_node()
+        raw_data = raw_data_node.raw_data
+        for key in id_list:
+            raw_data = raw_data[key]
+        return raw_data
+
+    def modify(self, data, method="update", reload=False):
+        """
+        Modify node metadata (original data stored in file) via method.
+        method is any method what contains python dictionary (default: update)
+        and data is attribute of the method
+
+        set reload  to True to dynamically recreate the subtree,
+        theoretically it may have some side effects, use it carefully.
+        Or reload whole tree to ensure data are completely fine.
+        """
+
+        getattr(self.node_get_raw_data(), method)(data)
+        self.store()
+        if reload:
+            self.reload()
+        return self
+
+    def __data_cleanup(self):
+        for item in self.climb(whole=True):
+            item.data = dict()
+            item.sources = []
+
+    def reload(self):
+        raw_node, _ = self.get_raw_data_node()
+        source_file = raw_node.sources[-1]
+        raw_node.__data_cleanup()
+        raw_node.grow(os.path.dirname(source_file))
+        raw_node.inherit()
+
+    def store(self):
+        raw_data_node, _ = self.get_raw_data_node()
+        target_file = raw_data_node.sources[-1]
+        with open(target_file, "w") as fd:
+            fd.write(dict_to_yaml(raw_data_node.raw_data))
```

Comment:
This will need a lock, but I'll  address that in #97

Reply:
yep, locking of modification of files is another issue